# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/script/ScriptSettings.cpp
# Commit: a50f60a612d3
# Full Hash: a50f60a612d31de79369e59933242ee10a5d810f
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2022-02-18 09:08:22
# Regressor Bug: 1742437
# File Overlap Count: 1
# Description:
#   Bug 1742437 - move mElement from ScriptFetchOptions to DOMScriptLoadContext; r=jonco,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D136742
# ==============================================================================

diff -r 0ac6d9d26e2b -r a50f60a612d3 dom/script/ScriptSettings.cpp
--- a/dom/script/ScriptSettings.cpp	Thu Feb 17 15:57:48 2022 +0000
+++ b/dom/script/ScriptSettings.cpp	Thu Feb 17 15:57:48 2022 +0000
@@ -57,13 +57,9 @@
 
   LoadedScript* script = static_cast<LoadedScript*>(aPrivateValue.toPrivate());
 
-  if (!script->GetFetchOptions()) {
-    return nullptr;
-  }
-
   JS::Rooted<JS::Value> elementValue(aCx);
   {
-    nsCOMPtr<Element> domElement = script->GetFetchOptions()->mElement;
+    nsCOMPtr<Element> domElement = script->GetScriptElement();
     if (!domElement) {
       return nullptr;
     }
